flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ReadableCatalog.java

```diff
@@ -224,4 +224,54 @@ CatalogPartition getPartition(ObjectPath tablePath, CatalogPartitionSpec partiti
 	 */
 	boolean functionExists(ObjectPath functionPath) throws CatalogException;
 
+	// ------ Statistics ------
+
+	/**
+	 * Get the statistics of a table.
+	 *
+	 * @param tablePath path of the table
+	 * @return the statistics of the given table
+	 *
+	 * @throws TableNotExistException if the table does not exist in the catalog
+	 * @throws CatalogException	in case of any runtime exception
+	 */
+	CatalogTableStatistics getTableStatistics(ObjectPath tablePath) throws TableNotExistException, CatalogException;
+
+	/**
+	 * Get the column statistics of a table.
+	 *
+	 * @param tablePath path of the table
+	 * @return the column statistics of the given table
+	 *
+	 * @throws TableNotExistException if the table does not exist in the catalog
+	 * @throws CatalogException	in case of any runtime exception
+	 */
+	CatalogColumnStatistics getTableColumnStatistics(ObjectPath tablePath) throws TableNotExistException, CatalogException;
```

Comment:
What does this API try to accomplish?  If this is for all columns of the table, should it return a list of `CatalogColumnStatistics`? If it's for a single column, should it take a column name?

Besides, what happens if this API is called on a partitioned table?

Reply:
good thinking. Let me rethink this. It seems to me that column stats of a table should be a list of stats for all columns. This probably requires some rework. Stay tuned.

Reply:
Called on partitioned table is fine, though it may not exist or be accurate.

Reply:
ok, no problem. Please also take the APIs in ReadableWritableCatalog into consideration